From 2494b030ba9334c7dd7df9b9f7abe4eacc950ec5 Mon Sep 17 00:00:00 2001
From: Fenghua Yu <fenghua.yu@intel.com>
Date: Tue, 17 May 2011 12:33:26 -0700
Subject: x86, cpufeature: Fix cpuid leaf 7 feature detection
Patch-mainline: v3.0-rc1
References: FATE#311936 bnc#698793

CPUID leaf 7, subleaf 0 returns the maximum subleaf in EAX, not the
number of subleaves.  Since so far only subleaf 0 is defined (and only
the EBX bitfield) we do not need to qualify the test.

Signed-off-by: Fenghua Yu <fenghua.yu@intel.com>
Link: http://lkml.kernel.org/r/1305660806-17519-1-git-send-email-fenghua.yu@intel.com
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
Cc: <stable@kernel.org> 2.6.36..39
Signed-off-by: Youquan Song <youquan.song@intel.com>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Automatically created from "patches.arch/repmovs-06-fix-cpuid-leaf-7-feature-detection.patch" by xen-port-patches.py

--- sle11sp2-2011-06-28.orig/arch/x86/kernel/cpu/common-xen.c	2011-06-28 13:13:14.000000000 +0200
+++ sle11sp2-2011-06-28/arch/x86/kernel/cpu/common-xen.c	2011-06-28 13:13:21.000000000 +0200
@@ -588,8 +588,7 @@ void __cpuinit get_cpu_cap(struct cpuinf
 
 		cpuid_count(0x00000007, 0, &eax, &ebx, &ecx, &edx);
 
-		if (eax > 0)
-			c->x86_capability[9] = ebx;
+		c->x86_capability[9] = ebx;
 	}
 
 	/* AMD-defined flags: level 0x80000001 */
